### PR TITLE
38/delete user by

### PR DESCRIPTION
**Pull request by nicksantoro**
route for deleting user by user's id

### CODE REVIEW

File: queries/users.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const knex = require('./db');
+
+const getAllUsers = () => {
+  return knex('users').orderBy('created_at', 'desc');
+}
+
+const getUserById = (userId) => {
+  console.log(userId)
```
Comment (developerX) [inline queries/users.js]: Remove Console log

File: queries/users.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return knex('users')
+    .where('id', userId).first()
+    .then(result => {
+      return result;
+    })
+}
+
+const deleteUserById = (userId) => {
+  console.log(userId)
```
Comment (developerX) [inline queries/users.js]: Remove console.log

File: src/models/users.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const userQuery = require('../../queries/users');
+
+const getAllUsers = () => {
+  const user = userQuery.getAllUsers();
+  return user.then(result => result.length < 1 ? { error: 'error retrieving users', status: 500 } : result)
```
Comment (developerX) [inline src/models/users.js]: Does this error mean that there was an issue retrieving users? What about if there are no users?

File: src/models/users.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const getUserById = (userId) => {
+  const user = userQuery.getUserById(userId);
+  return user.then(result => !result ? { error: "error retrieving user by id", status: 500 } : result)
```
Comment (developerX) [inline src/models/users.js:10]: Doesn't this mean there was no user with that Id?

**Review by developerX — CHANGES_REQUESTED** [review]
Please answer the questions?

File: src/models/users.js (hunk 4 above)
Comment (nicksantoro) [reply src/models/users.js:10]: issue corrected, sending again for approval

File: queries/users.js (hunk 2 above)
Comment (nicksantoro) [reply queries/users.js]: issue corrected

File: queries/users.js (hunk 1 above)
Comment (nicksantoro) [reply queries/users.js]: issue corrected

File: src/models/users.js (hunk 3 above)
Comment (nicksantoro) [reply src/models/users.js]: updated error message, resending for approval

**Review by developerX — APPROVED** [review]
Thank you!

**Pull request merged**